### PR TITLE
fix(button): Trigger changes to the buttongroup when a button is deselected

### LINKED ISSUE #13801: IgxButton: Updating the value bound to the selected prop doesn't deselect the button
## Description  
When updating the value bound to the `selected` property of an `IgxButton`, I need to manually call the `deselectButton` method.

Looking at the igniteui code, it seems that buttons only trigger changes to its buttongroup owner when they are selected.
When they're deselected by updating the model, the buttongroup is not notified. 
In this case, the model and the visual state shown by the component looks disconnected.

image

The selection state is only updated or notified on init, or when a button changes from deselected to selected, not the other way around.

The issue was first reproduced in AppBuilder

 * igniteui-angular version: 17.0.6
 * browser: Chrome 120.0.6099.217

## Steps to reproduce  

1. Create a button group with several buttons
2. Bind the `selected` prop of the buttons to a certain value.
3. Programmatically update the value (e.g. implement a reset logic to the form that the buttons are part of)

## Result  
After updating the value for the `selected` prop (e.g. hitting the reset button in the form), the selected buttons continue being selected, and I need to manually call `deselectButton` to de-select them.

## Expected result  
After updating the value for the `selected` prop (e.g. hitting the reset button in the form), the selected buttons should turn to de-selected without having to call the `deselectButton` method.

### PR DESCRIPTION
**Pull request by PlamenD95**
Closes #13801   

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/buttonGroup/buttongroup.component.spec.ts [hunk 1]
```diff
@@ -358,6 +360,19 @@ describe('IgxButtonGroup', () => {
...
+    it('should style the corresponding button as deselected when the value bound to the selected input changes', () => {
+        const fixture = TestBed.createComponent(ButtonGroupButtonWithBoundSelectedOutputComponent);
+        fixture.detectChanges();
+        
+        const btnGroupInstance = fixture.componentInstance.buttonGroup;
+
+        fixture.componentInstance.selectedValue = 100;
+        fixture.detectChanges();
+
+        expect(btnGroupInstance.selectedButtons.length).toBe(0);
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/buttonGroup/buttongroup.component.spec.ts:372]: To indicate that a change in selection really happened, it is good to verify that initially selectedButtons.lenght is > 0 and buttons[1] is selected.

File: projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts [hunk 2]
```diff
@@ -366,22 +366,27 @@ export class IgxButtonGroupComponent extends DisplayDensityBase implements After
...
+            // deselect other buttons if selectionMode is not multi
+            if (this.selectionMode !== 'multi' && this.selectedIndexes.length > 1) {
+                this.buttons.forEach((_, i) => {
+                    if (i !== index && this.selectedIndexes.indexOf(i) !== -1) {
+                        this.deselectButton(i);
+                    }
+                });
+            }
+        } else {
+            this.deselectButton(index);
```
[the remark this answers is not in the document]
Comment (PlamenD95) [reply projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts:387]: I believe that calling `deselectButton` is correct in this case:

image

File: projects/igniteui-angular/src/lib/buttonGroup/buttongroup.component.spec.ts (hunk 1 above)
Comment (PlamenD95) [reply projects/igniteui-angular/src/lib/buttonGroup/buttongroup.component.spec.ts:375]: Okay, I've added this suggestion, thank you.

**Review by hanastasov — DISMISSED** [review]

**Comment by hanastasov** [comment]
Please add a note in the CHANGELOG about this behavioral change.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ All notable changes for each version of this project will be documented in this
 - `IgxPivotGrid`
     - `IPivotDimension` interface now exposes a property called `displayName` similar to the one in the `IPivotValue` interface. This property is optional and will be displayed inside the chips for rows and columns in the `IgxPivotGrid`. If the `displayName` proeprty is not set, `memberName` will be used as a fallback.
 - New directive -  `igxIconButton` directive that provides a way to use an icon as a fully functional button has been added. The new `igxIconButton` comes in three types - flat, outlined and contained (default). All `igxButton`'s with type `icon` will be automatically migrated to the new `igxIconButton`'s with `ng update`.
+- `IgxButton`
+    - **Behavioral Change** When updating the value bound to the `selected` input of an `IgxButton`, the button group in which the button resides is now able to update the styling of the button from selected to deselected.
```
Comment (hanastasov) [inline CHANGELOG.md:15]:
```suggestion
    - **Behavioral Change** `buttonSelected` event is now emitted not only when a button gets selected, but also when it gets unselected. A benefit of that is when updating the value bound to the `selected` input of an `IgxButton`, the button group in which the button resides is now able to update the styling of the button from selected to deselected. If this event was used in a scenario where it is assumed that the button gets selected, is a good idea the logic to be branched now based on `eventArgs.selected` condition.
```

**Review by hanastasov — APPROVED** [review]

**Comment by hanastasov** [comment]
@simeonoff if there is a `buttonSelected` event it makes sense for `buttonDeselected` to exist too. The naming would better be `selected/deselected` but this is another story.

For now, I think going with the same name and not add a new one to the api is ok. One can always base a custom logic based on `eventArgs.seelected` value.

I am OK with this change, if you are OK too, we we can proceed and merge it.

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,8 @@ All notable changes for each version of this project will be documented in this
...
     - `IPivotDimension` interface now exposes a property called `displayName` similar to the one in the `IPivotValue` interface. This property is optional and will be displayed inside the chips for rows and columns in the `IgxPivotGrid`. If the `displayName` proeprty is not set, `memberName` will be used as a fallback.
 - New directive -  `igxIconButton` directive that provides a way to use an icon as a fully functional button has been added. The new `igxIconButton` comes in three types - flat, outlined and contained (default). All `igxButton`'s with type `icon` will be automatically migrated to the new `igxIconButton`'s with `ng update`.
+- `IgxButton`
+    - **Behavioral Change** `buttonSelected` event is now emitted not only when a button gets selected, but also when it gets unselected. A benefit of that is when updating the value bound to the `selected` input of an `IgxButton`, the button group in which the button resides is now able to update the styling of the button from selected to deselected. If this event was used in a scenario where it is assumed that the button gets selected, is a good idea the logic to be branched now based on `eventArgs.selected` condition.
```
Comment (simeonoff) [inline CHANGELOG.md:15]:
```suggestion
    - **Behavioral Change** `buttonSelected` event is now emitted not only when a button gets selected, but also when it gets deselected. A benefit of that is when updating the value bound to the `selected` input of an `IgxButton`, the button group in which the button resides is now able to update the styling of the button from selected to deselected. If this event was used in a scenario where it is assumed that the button gets selected, it's a good idea the logic to be branched now based on `eventArgs.selected` condition.
```

**Review by simeonoff — CHANGES_REQUESTED** [review]
See comment.

**Review by hanastasov — APPROVED** [review]

**Pull request merged**